App/PyImports/ProxyPyQml.py

```diff
@@ -62,6 +63,131 @@ def loadPhasesFromFile(self):
         # explicit emit required for the view to reload the model content
         self.projectChanged.emit()
         self.onProjectUnsaved()
+    
+    @Slot(str)
+    def loadExperiment(self, selectedNameFilter):
+        """
+        Selects the appropriate loading algorithm
+        """
+    
+        if "(*.cif)" in selectedNameFilter:
+            self.loadExperimentFromFile()
+        elif "(*.xye)" in selectedNameFilter:
+            self.loadExperimentXYE()
+        else:
+            raise IOError("Given selectedNameFilter not handled in loadExperiment.")
+
+    @Slot()
+    def loadExperimentXYE(self):
+        """
+        Loads non cif data files, adds fake cif information, and loads
+        """
+        
+        self._experiment_rcif_path = self._project_control.experiment_rcif_path
+        if "file://" in self._experiment_rcif_path[0:8]:
+            experiment_rcif_path = self._experiment_rcif_path.split("file://", 1)[1]
+        else:
+            experiment_rcif_path = self._experiment_rcif_path
+
+        data = np.loadtxt(experiment_rcif_path)
+        joined = [" ".join(item) for item in data.astype(str)]
+        data_string = "\n".join(joined)
+        
+        unpolarized_cif_start = """
+data_pd
+
+_setup_wavelength      2.00
+_setup_offset_2theta   0.00
+
+_pd_instr_resolution_u  0.15000
+_pd_instr_resolution_v -0.30000
+_pd_instr_resolution_w  0.30000
+_pd_instr_resolution_x  0.00000
+_pd_instr_resolution_y  0.15000
+
+loop_
+_pd_background_2theta
+_pd_background_intensity
+ 0.0000         10.0
+ 180.0000       10.0
+ 
+loop_
+_phase_label
+_phase_scale
+_phase_igsize
+PbSO4 1.1328 0.0
+
+loop_
+_pd_meas_2theta
+_pd_meas_intensity
+_pd_meas_intensity_sigma
+"""
+  
+        polarized_cif_start = """
+data_pd
+
+_setup_wavelength      0.84
+_setup_field           1.00
+_setup_offset_2theta   0.00
+
+_diffrn_radiation_polarization -0.87
+_diffrn_radiation_efficiency    1.00
+
+_pd_instr_resolution_u 15.00
+_pd_instr_resolution_v -3.00
+_pd_instr_resolution_w  0.60
+_pd_instr_resolution_x  0.00
+_pd_instr_resolution_y  0.00
+
+_range_2theta_min     4.000
+_range_2theta_max    80.000
+
+loop_
+_exclude_2theta_min
+_exclude_2theta_max
+0.0 1.0
+
+loop_
+_pd_background_2theta
+_pd_background_intensity
+ 4.5 256.0
+40.0 158.0
+80.0  65.0
+
+loop_
+_phase_label
+_phase_scale
+_phase_igsize
+Fe3O4 0.02381 0.0
```

Comment:
Experiment is loaded after the phase, so the phase name is already known. Could we replace this hard coded name `Fe3O4` with the name of already loaded phase using `phasesIds()[0]` before merging into `cif_string`?

Reply:
Done 👍 Done with a replace command in ProxyPyQml layer.